### DISCUSSION TITLE
Welcome to blade Discussions!

### DISCUSSION THREAD

**Discussion by mcfriend99** [zuri-lang/zuri#35] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mcfriend99** [comment]
Hi all,

You can follow developments of modules in the [`modules`](https://github.com/blade-lang/blade/tree/modules) branch.

Regards

**Comment by ibnz36** [comment]
Are you going to re-write all the parser on blade?

**Reply by mcfriend99** [reply]
Blade currently use a single pass compiler which doesn’t use ASTs and to expose Blade to developers and give the ability to develop such tools as automatic documentation generation, syntax highlighters and similar utilities, it needs to be exposed as ASTs. Another alternative is to switch to an AST instead of a single pass compiler and expose it as a module but this means a complex and bloated implementation for a simple language. So the AST module is basically rewriting the parser with as recursive descent in Blade.

**Reply by mcfriend99** [reply]
Do you have an alternative to propose??

**Reply by ibnz36** [reply]
> Blade currently use a single pass compiler which doesn’t use ASTs and to expose Blade to developers and give the ability to develop such tools as automatic documentation generation, syntax highlighters and similar utilities, it needs to be exposed as ASTs. Another alternative is to switch to an AST instead of a single pass compiler and expose it as a module but this means a complex and bloated implementation for a simple language. So the AST module is basically rewriting the parser with as recursive descent in Blade.

Ok, that makes sense
